pkg/config/command_provider.go

```diff
@@ -0,0 +1,55 @@
+package config
+
+import (
+	"io"
+	"os/exec"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	commandPasswordType = "command"
+)
+
+var _ PasswordProvider = (*CommandPasswordProvider)(nil)
+
+type CommandPasswordProvider struct {
+	Command   string   `mapstructure:"command"`
+	Arguments []string `mapstructure:"arguments"`
+}
+
+func (r CommandPasswordProvider) GetPassword() (string, error) {
+	cmd := exec.Command(r.Command, r.Arguments...)
```

Comment:
Shouldn't we improve the security of this method by allowing only certain commands?

```go
    cmdWhitelist := map[string]bool{
        "echo": true,
        "cat":  true,
        // Ajouter d'autres commandes autorisées ici
    }
    if !cmdWhitelist[r.Command] {
        return "", fmt.Errorf("command '%s' not allowed", r.Command)
    }
```

Reply:
I personally am not a fan of this. It would require us to list all possible password managers binaries available. 
Also allowing new binaries would require a code change. 

As for the security implication, I see this attack vector only when running an untrusted config someone shared
with you. This doesn't seem like a likely scenario in my head. Hence the implications of only allowing a certain set
of commands out-weights the gain in my mind.